keycloak-iam/src/test/kotlin/com/valtimo/keycloak/KeycloakRoleServiceTest.kt

```diff
@@ -0,0 +1,76 @@
+package com.valtimo.keycloak
+
+import com.valtimo.keycloak.service.KeycloakRoleService
+import com.valtimo.keycloak.service.KeycloakService
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.keycloak.admin.client.Keycloak
+import org.keycloak.admin.client.KeycloakBuilder
+import org.keycloak.representations.idm.RoleRepresentation
+import org.mockito.Answers
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+import kotlin.test.assertEquals
+
+@ExtendWith(MockitoExtension::class)
+internal class KeycloakRoleServiceTest {
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    lateinit var keycloakService: KeycloakService
+
+    @InjectMocks
+    lateinit var keycloakRoleService: KeycloakRoleService
+
+    @Test
+    fun `should return all roles found when prefix is null`() {
+
+        // given
+        defaultConditions()
+
+        // when
+        val result = keycloakRoleService.findRoles(null)
+
+        // then
+        assertEquals(TEST_ROLE_NO_PREFIX_LIST, result)
+    }
+
+    @Test
+    fun `should return roles with TEAM in the name`() {
+
+        // given
+        defaultConditions()
+
+        // when
+        val result = keycloakRoleService.findRoles("ROLE_TEAM_")
+
+        // then
+        assertEquals(TEST_ROLE_LIST, result)
+    }
+
+    private fun defaultConditions() {
+        whenever(keycloakService.keycloak()).thenReturn(keycloak())
```

Comment:
You could've also mocked the Keycloak instance, since it's not called directly in your code. 

```suggestion
        whenever(keycloakService.keycloak()).thenReturn(mock<Keycloak>())
```

And you wouldn't need the `keycloak()` method below.

Reply:
Nice! have to get this new mocking into my system :) i changed it